ui/pages/confirmations/hooks/useConfirmationNavigation.ts

```diff
@@ -0,0 +1,147 @@
+import { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { ApprovalType } from '@metamask/controller-utils';
+import { isEqual } from 'lodash';
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { Json } from '@metamask/utils';
+import { TEMPLATED_CONFIRMATION_APPROVAL_TYPES } from '../confirmation/templates';
+import {
+  CONFIRM_ADD_SUGGESTED_NFT_ROUTE,
+  CONFIRM_ADD_SUGGESTED_TOKEN_ROUTE,
+  CONFIRM_TRANSACTION_ROUTE,
+  CONFIRMATION_V_NEXT_ROUTE,
+  CONNECT_ROUTE,
+  DECRYPT_MESSAGE_REQUEST_PATH,
+  ENCRYPTION_PUBLIC_KEY_REQUEST_PATH,
+  SIGNATURE_REQUEST_PATH,
+} from '../../../helpers/constants/routes';
+import { isSignatureTransactionType } from '../utils';
+import {
+  getApprovalFlows,
+  pendingApprovalsSortedSelector,
+} from '../../../selectors';
+
+const CONNECT_APPROVAL_TYPES = [
+  ApprovalType.WalletRequestPermissions,
+  'wallet_installSnap',
+  'wallet_updateSnap',
+  'wallet_installSnapResult',
+];
+
+export function useConfirmationNavigation() {
+  const confirmations = useSelector(pendingApprovalsSortedSelector, isEqual);
```

Comment:
When/how were all confirmations added to `pendingApprovalsSortedSelector`? How is that state populated?

Reply:
That's reading the approval requests directly from the `ApprovalController` state.

Reply:
Ah! Those were always there but weren't used for the queueing - got it!